ArchiSteamFarm/www/pages/config.html

```diff
@@ -170,15 +170,119 @@ <h4><i class="icon fas fa-info"></i> Not implemented</h4>
+                var hoursUntilCardDrops = ($("#HoursUntilCardDrops").val() !== '') ? parseInt($("#HoursUntilCardDrops").val()) : 3;
+                var passwordFormat = ($("#PasswordFormat").val() !== '') ? parseInt($("#PasswordFormat").val()) : 0;
+                var sendTradePeriod = ($("#SendTradePeriod").val() !== '') ? parseInt($("#SendTradePeriod").val()) : 0;
+                var steamLogin = ($("#SteamLogin").val() !== '') ? $("#SteamLogin").val() : null;
+                var steamMasterClanID = ($("#SteamMasterClanID").val() !== '') ? Number($("#SteamMasterClanID").val()) : 0;
+                var steamParentalPIN = ($("#SteamParentalPIN").val() !== '') ? $("#SteamParentalPIN").val() : "0";
+                var steamPassword = ($("#SteamPassword").val() !== '') ? $("#SteamPassword").val() : null;
+                var tradingPreferences = ($("#TradingPreferences").val() !== '') ? parseInt($("#TradingPreferences").val()) : 0;
+                var redeemingPreferences = ($("#RedeemingPreferences").val() !== '') ? parseInt($("#RedeemingPreferences").val()) : 0;
+
+                var gamesPlayedWhileIdle = [];
+                if ($("#GamesPlayedWhileIdle").val() !== '') {
+                    var value = $("#GamesPlayedWhileIdle").val();
+                    var appIDs = value.split(',');
+
+                    gamesPlayedWhileIdle = appIDs.map(Number);
+                }
+
+                var lootableTypes = [];
+                if ($("#LootableTypes").val() !== '') {
+                    var value = $("#LootableTypes").val();
+                    var lootTypes = value.split(',');
+
+                    lootableTypes = lootTypes.map(Number);
+                }
+
+                var matchableTypes = [];
+                if ($("#MatchableTypes").val() !== '') {
+                    var value = $("#MatchableTypes").val();
+                    var matchTypes = value.split(',');
+
+                    matchableTypes = matchTypes.map(Number);
+                }
+
+                var steamUserPermissions = {};
+                if ($("#SteamUserPermissions").val() !== '') {
+                    var permissions = [];
+                    var lines = $("#SteamUserPermissions").val().split('\n');
+
+                    for (var i = 0; i < lines.length; i++) {
+                        if (lines[i] !== '') {
+                            permissions.push(lines[i].split(':'));
+                        }
+                    }
 
-            var botConfig = JSON.stringify(objBotConfig);
+                    for (var j = 0; j < permissions.length; j++) {
+                        var obj = permissions[j];
+                        steamUserPermissions[obj[0]] = parseInt(obj[1]);
+                    }
+                }
+
+                var objBotConfig = {
+                    "BotConfig": {
+                        "AcceptGifts": acceptGifts,
```

Comment:
Hardcoding current properties of botconfig is absolutely unacceptable, this is what /Api/Structure and /Api/Type should be used for and should be done fully dynamic.

Reply:
Dont worry - it will be fully dynamic like `generateConfigChangerHTML()` and `loadConfigValuesForBot()`.

I just hardcoded it for now since I was still testing things. Next step will be to make that function dynamic.

Reply:
It should be dynamic now. Check it out if you want!